stackslib/src/net/dns.rs

```diff
@@ -292,18 +293,15 @@ impl DNSClient {
         loop {
             match self.requests_rx.try_recv() {
                 Ok(resp) => {
-                    if self.requests.contains_key(&resp.request) {
+                    if let Entry::Occupied(mut e) = self.requests.entry(resp.request.to_owned()) {
```

Comment:
I'd recommend `get_mut()` here, it's cleaner if you're modifying occupied entries:

```suggestion
                    if let Some(e) = self.requests.get_mut(resp.request) {
```